src/components/TradeWidget/index.tsx

```diff
@@ -196,7 +201,14 @@ const TradeWidget: React.FC = () => {
         validUntil,
         txOptionalParams: {
           onSentTransaction: (txHash: string): void => {
+            // reset form on successful order placing
+            reset(DEFAULT_FORM_STATE)
+            setUnlimited(false)
+            // unblock form
+            setIsSubmitting(false)
```

Comment:
That's a lot of setState, wrap in unstable_batch

Reply:
it's just 2.. also does unstable batch do anything for state hooks spread across different hooks?

Reply:
done! worked like a charm